common/src/main/java/org/jboss/hal/testsuite/creaper/command/AbstractTransportConfigAddCommand.java

```diff
@@ -0,0 +1,230 @@
+package org.jboss.hal.testsuite.creaper.command;
+
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An abstract class which implements common properties of connector and acceptor in messaging connections.
+ */
+abstract class AbstractTransportConfigAddCommand implements OnlineCommand {
+
+    protected TransportConfigType type;
+    protected String name;
+    protected String serverName;
+    protected Map<String, String> params;
+    protected boolean replaceExisting;
+
+    //in-vm
+    protected Integer serverId;
+
+    //generic and remote
+    protected String socketBinding;
+
+    //remote
+    protected String factoryClass;
+
+    protected AbstractTransportConfigAddCommand(InVmBuilder builder) {
```

Comment:
Can you parametrize references to generic type like this with wildcard? 
E.g. `InVmBuilder<?>`